### ISSUE TITLE
Support for NodeMCU-devkit 0.9 on Mac?

### ISSUE THREAD

**Issue by abl** [themadinventor/esptool#26]
Hi! I've got a couple of NodeMCU Devkits of the 0.9 rev (so the CH340G bridge) and I'm using the signed driver from mac-usb-serial.com. Out of the box, esptool isn't working.

These changes seem useful:

    self._port.setRTS(True)
    self._port.setDTR(False)
    time.sleep(0.1)
    self._port.setRTS(False)
    self._port.setDTR(True)
    time.sleep(0.1)

Doing this causes the ESP to boot in to mode (1,6) - user firmware boot identifies itself as (3,6).

The boot header is ```ets Jan  8 2013,rst cause:2, boot mode:(3,6)``` so it's a somewhat older firmware. I can't find any docs on what the second number means, but all of my working devices show a 7...

Any thoughts? :)

**Comment by abl** [comment]
Relying on USB for power, and connecting a different serial device, I boot in to (1,7) and was able to reflash without any issues. Even more mystified now...

**Comment by MacDada** [comment]
* I've had AT commands working, but it stopped when I did `AT+CIUPDATE`.
* I can't get any CH340G driver to work on Yosemite.
* I installed ubuntu with vagrant+virtualbox to make sure no OSX–specific software problems.
* I power NodeMcu from Mac using built in USB.
* I "talk" with NodeMcu through Arduino Mega serial with it's Atmega chip shut down – it speaks "garbage" (because of `AT+CIUPDATE`).
* Flashing with `esptool` doesn't work.
* I tried both with and without voltage divider.

@abl I show my setup plus what works and what doesn't here (sorry for my bad english :P): https://www.youtube.com/watch?v=AibvuZ4YB7Y
